File: solution_zzj/540._Single_Element_in_a_Sorted_Array.cc
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        if (nums.size()==1) 
            return nums[0];
        int l = 1, r = nums.size() - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[mid - 1] == nums[mid]) {
                if (mid % 2 == 0) {
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            } else {
                if (mid % 2 == 0) {
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }
        }
        return nums[r];
    }
};

File: Weijia/540. Single Element in a Sorted Array.py
class Solution(object):
    def singleNonDuplicate(self, nums):
        s1,s2=0,len(nums)-1
        while s1<s2:
            m=(s1+s2)/2
            if m%2!=0:
                m=m-1
            if nums[m]==nums[m+1]:
                s1=m+2
            else:
                s2=m
        return nums[s1]
